modules/programs/kitty.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.kitty;
+
+  eitherStrBoolInt = with types; either str (either bool int);
+
+  optionalPackage = opt:
+    optional (opt != null && opt.package != null) opt.package;
+
+  fontType = (import ../lib/font-type.nix { inherit lib; }).fontType;
+
+  toKittyConfig = generators.toKeyValue {
+    mkKeyValue = key: value:
+    let
+      value' =
+        if isBool value then (if value then "yes" else "no")
+        else toString value;
+    in
+      "${key} ${value'}";
+  };
+
+  toKittyKeybindings = generators.toKeyValue {
+    mkKeyValue = key: command: "map ${key} ${command}";
+  };
+in
+
+{
+  options.programs.kitty = {
+    enable = mkEnableOption "Kitty terminal emulator";
+
+    config = mkOption {
+      type = types.attrsOf eitherStrBoolInt;
+      default = {};
+      example = literalExample ''
+        {
+          scrollback_lines = 10000;
+          enable_audio_bell = false;
+          update_check_interval = 0;
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/kitty/kitty.conf</filename>. See
+        <link xlink:href="https://sw.kovidgoyal.net/kitty/conf.html" />
+        for the documentation.
+      '';
+    };
+
+    font = mkOption {
+      type = fontType;
+      default = null;
+      description = "The font to use.";
+    };
+
+    keybindings = mkOption {
+      type = types.attrsOf types.str;
+      default = {};
+      description = "Mapping of keybindings to actions.";
+      example = literalExample ''
+        {
+          "ctrl+c" = "copy_or_interrupt";
+          "ctrl+f>2" = "set_font_size 20";
+        }
+      '';
+    };
+
+    extraConfig = mkOption {
+      default = "";
+      type = types.str;
+      description = "Additional configuration to add.";
+      example = "";
```

Comment:
An empty example is not very useful, should be either removed or given some content.